### ISSUE TITLE
Low mAP of pretrained Faster-RCNN

### ISSUE THREAD

**Issue by gladcolor** [KaihuaTang/Scene-Graph-Benchmark.pytorch#87]
## ❓ Questions and Help
Used the pretrained Faster RCNN for to train SGDet, get very low mAP (0.0003) when evaluating the bbox. Could you tell me what is the issue? Had tried to redownload the pre-trained model from your repository.

100%|##########| 1250/1250 [16:41<00:00,  1.25it/s]
2020-09-23 11:10:26,403 maskrcnn_benchmark INFO: Total run time: 0:16:41.225906 (0.8009807245254517 s / img per device, on 4 devices)
2020-09-23 11:10:26,403 maskrcnn_benchmark INFO: Model inference time: 0:12:46.151817 (0.6129214532852173 s / img per device, on 4 devices)
creating index...
index created!
Loading and preparing results...
Converting ndarray to lists...
(398878, 7)
0/398878
DONE (t=3.64s)
creating index...
index created!
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=79.92s).
Accumulating evaluation results...
DONE (t=21.74s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.004
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.006
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.006
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.002
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.006
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.007
2020-09-23 11:18:37,225 maskrcnn_benchmark INFO: 
====================================================================================================
Detection evaluation mAp=0.0003
====================================================================================================
SGG eval:     R @ 20: 0.0000;     R @ 50: 0.0000;     R @ 100: 0.0000;  for mode=sgdet, type=Recall(Main).
SGG eval:  ng-R @ 20: 0.0000;  ng-R @ 50: 0.0000;  ng-R @ 100: 0.0000;  for mode=sgdet, type=No Graph Constraint Recall(Main).
SGG eval:    zR @ 20: 0.0000;    zR @ 50: 0.0000;    zR @ 100: 0.0000;  for mode=sgdet, type=Zero Shot Recall.
SGG eval: ng-zR @ 20: 0.0000; ng-zR @ 50: 0.0000; ng-zR @ 100: 0.0000;  for mode=sgdet, type=No Graph Constraint Zero Shot Recall.
SGG eval:    mR @ 20: 0.0000;    mR @ 50: 0.0000;    mR @ 100: 0.0000;  for mode=sgdet, type=Mean Recall.
----------------------- Details ------------------------
(above:0.0000) (across:0.0000) (against:0.0000) (along:0.0000) (and:0.0000) (at:0.0000) (attached to:0.0000) (behind:0.0000) (belonging to:0.0000) (between:0.0000) (carrying:0.0000) (covered in:0.0000) (covering:0.0000) (eating:0.0000) (flying in:0.0000) (for:0.0000) (from:0.0000) (growing on:0.0000) (hanging from:0.0000) (has:0.0000) (holding:0.0000) (in:0.0000) (in front of:0.0000) (laying on:0.0000) (looking at:0.0000) (lying on:0.0000) (made of:0.0000) (mounted on:0.0000) (near:0.0000) (of:0.0000) (on:0.0000) (on back of:0.0000) (over:0.0000) (painted on:0.0000) (parked on:0.0000) (part of:0.0000) (playing:0.0000) (riding:0.0000) (says:0.0000) (sitting on:0.0000) (standing on:0.0000) (to:0.0000) (under:0.0000) (using:0.0000) (walking in:0.0000) (walking on:0.0000) (watching:0.0000) (wearing:0.0000) (wears:0.0000) (with:0.0000) 
--------------------------------------------------------
SGG eval: ng-mR @ 20: 0.0000; ng-mR @ 50: 0.0000; ng-mR @ 100: 0.0000;  for mode=sgdet, type=No Graph Constraint Mean Recall.
----------------------- Details ------------------------
(above:0.0000) (across:0.0000) (against:0.0000) (along:0.0000) (and:0.0000) (at:0.0000) (attached to:0.0000) (behind:0.0000) (belonging to:0.0000) (between:0.0000) (carrying:0.0000) (covered in:0.0000) (covering:0.0000) (eating:0.0000) (flying in:0.0000) (for:0.0000) (from:0.0000) (growing on:0.0000) (hanging from:0.0000) (has:0.0000) (holding:0.0000) (in:0.0000) (in front of:0.0000) (laying on:0.0000) (looking at:0.0000) (lying on:0.0000) (made of:0.0000) (mounted on:0.0000) (near:0.0000) (of:0.0000) (on:0.0000) (on back of:0.0000) (over:0.0000) (painted on:0.0000) (parked on:0.0000) (part of:0.0000) (playing:0.0000) (riding:0.0000) (says:0.0000) (sitting on:0.0000) (standing on:0.0000) (to:0.0000) (under:0.0000) (using:0.0000) (walking in:0.0000) (walking on:0.0000) (watching:0.0000) (wearing:0.0000) (wears:0.0000) (with:0.0000) 
--------------------------------------------------------

...

2020-09-23 11:18:45,881 maskrcnn_benchmark INFO: -------------------------------
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 32768.0
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 32768.0
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 16384.0
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 16384.0
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 16384.0
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 16384.0
2020-09-23 11:25:24,474 maskrcnn_benchmark INFO: eta: 1 day, 3:50:23  iter: 200  loss: 1.6483 (2.3342)  auxiliary_ctx: 0.1947 (0.4157)  auxiliary_frq: 0.2053 (0.2132)  auxiliary_vis: 0.2256 (0.3120)  loss_refine_obj: 0.6943 (1.0113)  loss_rel: 0.3167 (0.3819)  time: 1.9830 (2.0125)  data: 0.0207 (0.0359)  lr: 0.109968  max mem: 6490
2020-09-23 11:32:04,596 maskrcnn_benchmark INFO: eta: 1 day, 3:38:45  iter: 400  loss: 1.5293 (1.9492)  auxiliary_ctx: 0.1900 (0.3062)  auxiliary_frq: 0.2097 (0.2131)  auxiliary_vis: 0.2051 (0.2664)  loss_refine_obj: 0.5935 (0.8091)  loss_rel: 0.3094 (0.3544)  time: 1.9549 (2.0066)  data: 0.0240 (0.0316)  lr: 0.196368  max mem: 6490
2020-09-23 11:38:42,789 maskrcnn_benchmark INFO: eta: 1 day, 3:27:47  iter: 600  loss: 1.4639 (1.8075)  auxiliary_ctx: 0.1762 (0.2673)  auxiliary_frq: 0.2090 (0.2132)  auxiliary_vis: 0.1982 (0.2495)  loss_refine_obj: 0.5870 (0.7357)  loss_rel: 0.2783 (0.3419)  time: 1.9449 (2.0014)  data: 0.0228 (0.0295)  lr: 0.240000  max mem: 6490
2020-09-23 11:45:24,081 maskrcnn_benchmark INFO: eta: 1 day, 3:22:09  iter: 800  loss: 1.3665 (1.7205)  auxiliary_ctx: 0.1594 (0.2453)  auxiliary_frq: 0.1998 (0.2125)  auxiliary_vis: 0.1833 (0.2393)  loss_refine_obj: 0.5260 (0.6916)  loss_rel: 0.2805 (0.3318)  time: 1.9940 (2.0026)  data: 0.0223 (0.0285)  lr: 0.240000  max mem: 6490
2020-09-23 11:52:02,228 maskrcnn_benchmark INFO: eta: 1 day, 3:13:32  iter: 1000  loss: 1.4452 (1.6644)  auxiliary_ctx: 0.1780 (0.2317)  auxiliary_frq: 0.2151 (0.2123)  auxiliary_vis: 0.2051 (0.2322)  loss_refine_obj: 0.5228 (0.6632)  loss_rel: 0.3092 (0.3250)  time: 1.9350 (2.0003)  data: 0.0255 (0.0284)  lr: 0.240000  max mem: 6490

**Comment by gladcolor** [comment]
It is getting better now after several rounds of training.

**Comment by bibekyess** [comment]
HI @gladcolor, for fasterrcnn, there is last_checkpoint file, so several rounds of training may work. But, for SG-Detection models like motif, there is no last_checkpoint, so several rounds of training is not working. Can you give me some hints on how you improved this? I mean I am also getting 0.0000 for all the predicates, so I want to know how you improved this? Thank you!
